### PR TITLE
Fix broken flag prop / new statusIcon prop

### LINKED ISSUE #3033: `flag` prop of k-card and k-list-item is broken
**Describe the bug**  
Commit 7ded08b89ef6066beafb243cf86bc23e0a129cce introduced a non-documented breaking change to the `flag` prop of `k-card` and `k-list-item` components.

The bug prevents using the `flag` prop in a generic manner as described [here](https://getkirby.com/docs/reference/plugins/ui/card#props) and [here](https://getkirby.com/docs/reference/plugins/ui/list-item#props). Instead, the `flag` prop is restricted/dedicated to page status indication since the generic `k-button` component has been replaced by the new and very specific `k-status-icon` component.

**To Reproduce** / **Expected behavior** 
A Vue component as the following one will not display the `trash` icon instead, the `circle` icon appears because of https://github.com/getkirby/kirby/blob/master/panel/src/components/Misc/StatusIcon.vue#L37.
```js
<k-list>
  <k-list-item key="item-a"
    :icon="{type:'calendar', back:'white'}
    :text:"Closer"
    :info:="28.02.2021"
    :flag="{icon: 'trash'}"
  />
</k-list>
```
image

**Kirby Version**  
3.5.0

### PR DESCRIPTION
**Pull request by bastianallgeier**
## Describe the PR

We've introduced a breaking change for the flag prop in list items and cards. This PR fixes this. It's based on https://github.com/getkirby/kirby/pull/3034

## Related issues

- Fixes #3033

### CODE REVIEW

File: panel/src/components/Layout/ListItem.vue [hunk 1]
```diff
@@ -17,12 +17,18 @@
...
+        v-else-if="statusIcon"
+        v-bind="statusIcon"
+        class="k-list-item-status"
+      />
       <slot name="options">
-        <k-status-icon
-          v-if="flag"
-          v-bind="flag"
-          class="k-list-item-status"
-        />
```
Comment (lukasbestle) [inline panel/src/components/Layout/ListItem.vue]: Why was the component moved out of the slot?

Comment (bastianallgeier) [reply panel/src/components/Layout/ListItem.vue]: Yikes, good catch!

**Comment by afbora** [comment]
`flag` working great but i can't test `statusIcon`. I have never used it before, missing something?

````vue
<k-list-item
  :statusIcon="{
    icon: 'alert',
    click: function() { alert('Hi!'); }
  }"
  text="Some image from unsplash"
/>

````

**Comment by bastianallgeier** [comment]
> `flag` working great but i can't test `statusIcon`. I have never used it before, missing something?
> 
> ```
> <k-list-item
>   :statusIcon="{
>     icon: 'alert',
>     click: function() { alert('Hi!'); }
>   }"
>   text="Some image from unsplash"
> />
> ```

camelCase props need to be passed with kebab-case in Vue.js:

```
<k-list-item
   :status-icon="{
     icon: 'alert',
     click: function() { alert('Hi!'); }
   }"
   text="Some image from unsplash"
/>
```

**Comment by afbora** [comment]
@bastianallgeier `status-icon` doesn't work for me 😞 
`click` handler working but icon always `circle.`

**Comment by bastianallgeier** [comment]
@afbora ah, sorry. The status icon component has different props than a simple button 

```
<k-list-item
   :status-icon="{
     status: 'listed',
     tooltip: 'some label',
     disabled: false,
     click: function() { alert('Hi!'); }
   }"
   text="Some image from unsplash"
/>
```

**Review by afbora — APPROVED** [review]
Awesome ❤️

File: panel/src/components/Layout/Card.vue [hunk 2]
```diff
@@ -19,12 +19,18 @@
...
+          v-if="flag"
+          v-bind="flag"
+          class="k-card-options-button"
+          @click="flag.click"
+        />
+        <k-status-icon
+          v-else-if="statusIcon"
+          v-bind="statusIcon"
+          class="k-card-options-button"
+        />
```
Comment (lukasbestle) [inline panel/src/components/Layout/Card.vue:33]: @bastianallgeier I don't know about the inner workings of these components, but in `Card` (unlike `ListItem`), the components *were* outside the options slot before. However I can imagine that it is correct like you did it now.

Comment (bastianallgeier) [reply panel/src/components/Layout/Card.vue:33]: Yes, I think it was wrong before. The idea of the slot is that you can replace the options/flag entirely with your own buttons. I.e. 

```
<k-list-item>
  <template slot="options">
    <my-own-flag />
    <my-own-dropdown />
  </template>
</k-list-item>
```

Comment (lukasbestle) [reply panel/src/components/Layout/Card.vue:33]: Yep, that's why I thought as well. 👍

**Pull request merged**